src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -1026,7 +1018,7 @@ bool FullyConnected::tryExtractParamForLLMFc(llmdnn::fc_create_param& param) {
             // compute q/dq
             auto p = getenv("USE_INT8_WEIGHT");
             if (p && p[0] == '1') {
-                auto weightPtr = getParentEdgeAt(1)->getMemoryPtr();
+                auto weightPtr = castMemoryPtr(getParentEdgeAt(WEIGHTS_ID)->getMemoryPtr(), data_type);
```

Comment:
We'd better to cache the weightPtr in order to reuse it
 in tryCompressWeightForLLMFc() because it's a time cost operation.